### DISCUSSION TITLE
Export model to external application

### DISCUSSION THREAD

**Discussion by iry47** [NVIDIA-NeMo/NeMo#2351] [Q&A]
Hi, I've been working with the punctuation and capitalization model, and training it on french data. After training the model and arriving at a checkpoint, what is the process of exporting the model so I can use it in my application?
`model.save` doesn't exist on this model (I get an error telling me that), I've saved the all the fiels in the nemo_experiments folder but I can't find any documentation on how to convert `.ckpt` files to a model.
Thanks in advance

**Comment by VahidooX** [comment]
All nemo models have a method to save the model as a nemo file which you may call by model.save_to(PATH).

**Reply by iry47** [reply]
Hello @VahidooX, thank you for the response. I've since found this function and been able to apply it, so success ! 
Now I'm trying to load my model for fine tunning and testing on my data, but install Nemo on my mac seems to be a big struggle. Any tips on how to solve the `ModeluNotFound: Nemo` ?
I've tried following all the installation instructions on the github and elsewhere, and honestly many times the urls/packages written come up as `not found`

**Reply by titu1994** [reply]
The only documentation that should be followed is the one linked to in the readme. Module not found can occur cause Nemo wasn't installed or installed in different conda env

**Reply by iry47** [reply]
Hi @titu1994, I've tried to install from Pip, Pip from source and From source without success. After trying the readme I searched other options and others in the nemo discussions that have had problems, but still no luck. 
For example, when I try `python -m pip install git+https://github.com/NVIDIA/NeMo.git@main#egg=nemo_toolkit[all]`, I get the error that no matching package can be found

**Reply by titu1994** [reply]
pip install nemo_toolkot[all] is the standard way to install on both Linux and windows. Could you try with a conda env to make sure it's being installed into the right environment

**Reply by bermeitinger-b** [reply]
This message indicates that you have an unsupported Python version.

**Reply by iry47** [reply]
Hi @titu1994 and @bermeitinger-b, thanks for the replies. I tried installing it into a conda env and still no luck. He's what I did:

`(base) ryanheadley@mbp-de-ryan nemo_test % conda create --name nemoTest
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /Users/ryanheadley/opt/anaconda3/envs/nemoTest

Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

(base) ryanheadley@mbp-de-ryan nemo_test % conda activate nemoTest
(nemoTest) ryanheadley@mbp-de-ryan nemo_test % pip install nemo_toolkit[all]
zsh: no matches found: nemo_toolkit[all]
(nemoTest) ryanheadley@mbp-de-ryan nemo_test % conda install -c nemo_toolkit[all]
zsh: no matches found: nemo_toolkit[all]
(nemoTest) ryanheadley@mbp-de-ryan nemo_test % python --version
Python 3.7.9`

**Reply by bermeitinger-b** [reply]
I guess your shell consumes the [] charaters. Can you try with 
`pip install "nemo_toolkit[all]"` (mind the quotation marks)

Same with the git source:
`python -m pip install "git+https://github.com/NVIDIA/NeMo.git@main#egg=nemo_toolkit[all]"`

**Reply by iry47** [reply]
Awesome, thanks so much! 
It found the package and started downloading but ran into an error about the compiler being at least C++11... It seems like my macOS Big sur should come with all of that automatically - have you seen this before?

clang: error: invalid version number in 'MACOSX_DEPLOYMENT_TARGET=11.2'
  error: command 'clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pesq

RuntimeError: Unsupported compiler -- at least C++11 support is needed!
  ----------------------------------------
  ERROR: Failed building wheel for pyamg

**Reply by titu1994** [reply]
@okuchaiev Did you face this on Mac installation ? Upgrading gcc should fix this right ?

**Reply by iry47** [reply]
@titu1994 I am downloading this on mac, and I upgraded my gcc to 10.1.0 which should be fine no? Although I still have the same error

**Reply by bermeitinger-b** [reply]
Your error message indicates that you're using clang instead of gcc. You should either set it to gcc or update clang as well.

**Reply by iry47** [reply]
I guess it uses clang automatically, but my clang is version 11.0.3 which apparently should have by default C++11 or higher, so why would I be erroring out? Do you know how to set it to gcc instead?

**Reply by iry47** [reply]
Thanks for all the help!
I found the error, for anyone running into the same problem with  [MacOS BigSur clang version](https://stackoverflow.com/questions/63972113/big-sur-clang-invalid-version-error-due-to-macosx-deployment-target)

Now that I've successfully been able to download Nemo, the module is not being found in my import. I've restarted everything, followed the docs and example colabs but no luck... any ideas?
